python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py

```diff
@@ -418,11 +414,25 @@ def _update_settings(
         kernel: "Kernel | None" = None,
     ) -> None:
         """Update the settings with the chat history."""
+
+        def _config_call_back(
```

Comment:
Similar to my previous question, why do we need this common code duplicated across multiple AI service related code?

Reply:
It's due to the settings type.

In addition, previously the function calling util had a dependency on the OpenAI setting, which is probably not what we want.

Reply:
The settings type can be the generic `PromptExecutionSettings` and we can look for the tools attribute, and then operate that way.